ast27/Python/ast.c

```diff
@@ -765,6 +765,12 @@ ast_for_arguments(struct compiling *c, const node *n)
     defaults = (n_defaults ? asdl_seq_new(n_defaults, c->c_arena) : NULL);
     if (!defaults && n_defaults)
         return NULL;
+    /* type_comments will be lazily initialized if needed.  If there are no
+       per-argument type comments, it will remain NULL.  Otherwise, it will be
+       an asdl_seq with length equal to the number of args (including varargs
+       and kwargs, if present) and with members set to the string of each arg's
+       type comment, if present, or NULL otherwise.
+     */
```

Comment:
Cool. I think there's a version of this also that belongs in the ASDL file, which is generally the reference for people wanting to consume the AST. It's not totally obvious what the semantics of this sequence is, particularly as it's quite different from the other parallel-ish-to-args sequence, `defaults`.